src/test/java/DockerUtilityTest.java

```diff
@@ -19,28 +26,41 @@ public static void setUp() throws InterruptedException {
                 DockerImage.MICROSOFT_SQL_SERVER_2019_LATEST,
                 DockerImage.MICROSOFT_SQL_SERVER_2017_CU21_UBUNTU_16_04,
                 DockerImage.MICROSOFT_SQL_SERVER_2017_LATEST,
-                DockerImage.MYSQL_LATEST
+                DockerImage.MYSQL_LATEST,
+                DockerImage.MONGODB_LATEST
         );
+
+        createdContainers = new HashMap<>();
         containersToCreate = Arrays.asList(
                 DockerImage.MICROSOFT_SQL_SERVER_2017_LATEST,
                 DockerImage.MICROSOFT_SQL_SERVER_2019_LATEST,
-                DockerImage.MYSQL_LATEST
+                DockerImage.MYSQL_LATEST,
+                DockerImage.MONGODB_LATEST
         );
+
         // containersToRun should be a subset of containersToCreate, as only if the container is created can it be run.
+        runningContainers = new HashMap<>();
         containersToRun = Arrays.asList(
                 DockerImage.MICROSOFT_SQL_SERVER_2019_LATEST,
-                DockerImage.MYSQL_LATEST
+                DockerImage.MYSQL_LATEST,
+                DockerImage.MONGODB_LATEST
         );
 
         dockerUtility = new DockerUtility();
         for (DockerImage imageToPull : imagesToPull) {
             dockerUtility.pullImage(imageToPull);
         }
         for (DockerImage containerToCreate : containersToCreate) {
-            dockerUtility.createContainer(containerToCreate, containerToCreate.toString(), true);
-        }
-        for (DockerImage containerToRun : containersToRun) {
-            dockerUtility.createAndRunContainer(containerToRun, containerToRun.toString(), true);
+            String containerName = containerToCreate.name() + "_" + containerToCreate.toString() + "_" + Instant.now().getEpochSecond();
```

Comment:
do you think it's better to name the container, in a more human readable way -
name_name_YYYY_MM_dd_hh_mm_ss vs name_name_<long number>?

Reply:
Honestly, doesn't matter. The usage of epoch was to ensure unique names, the actual time is not useful at that point considering we have logging for when we start/stop containers.

Reply:
👍